### ISSUE TITLE
add comparison to R package and scikit-learn

### ISSUE THREAD

**Issue by hugoguh** [glm-tools/pyglmnet#8]
make separate notebook to demonstrate comparison with R results
call R from python using os.system(‘Rscript .....

**Comment by titipata** [comment]
There is a library [`rpy2`](http://rpy2.bitbucket.org/) that we can use in notebook. We can install using `pip install rpy2`. However, I have issue with IPython kernel dying when I load it to the notebook `%load_ext rpy2.ipython`. I install python using Anaconda with Mac OSX. 

- `$ conda --version` > `conda 3.14.1`
- `$ which python` > `~/anaconda/bin/python`
- `$ which ipython` > `~/anaconda/bin/ipython`

Let me know if someone have run around for this issue.

**Comment by hugoguh** [comment]
it involves saving and loading files but can easily be done with something like this:
```python
def Rglmnet_poisson(Xr, Yr, Xt):
    scaler = StandardScaler().fit(Xr)
    Xr = scaler.transform(Xr)
    Xt = scaler.transform(Xt)

    X_train = pd.DataFrame(Xr, index=None)
    Y_train = pd.DataFrame(Yr, index=None)
    X_test = pd.DataFrame(Xt, index=None)

    #print "saving csv files"
    X_train.to_csv('X_train.csv',index=False)
    Y_train.to_csv('Y_train.csv',index=False)
    X_test.to_csv('X_test.csv',index=False)
    
    #print "running R"
    os.system('Rscript cv_glmnet_poisson.R')
    
    #print 'loading predictions'
    Yt = pd.read_csv('./Y_pred.csv')
    
    return np.exp(Yt['s0’])
```

It works, but it would be better to do it with `rpy2` like @titipata suggests

**Comment by jasmainak** [comment]
@hugoguh @pavanramkumar maybe it may also be nice to add comparison to `scikit-learn` in terms of features offered / benchmarks for timing / accuracy. It's not super clear to me yet what the differences / enhancements are (or can be).

**Comment by hugoguh** [comment]
got some weird results when trying pyglmnet with `distr=normal` on Roozbeh’s electro imaging data. Will build a comparison with `sickit-learn` soon to document that possible bug

**Comment by hugoguh** [comment]
ok the problem with fitting Roozbeh’s data is the learning rate. We stop getting very weird fits with smaller learning rates. 
still not very good fits though when compared with `sickit-learn`, probably needs adaptive learning rate

**Comment by themantalope** [comment]
What is the status of this issue? Also, just in general I have had less bugs and better control when calling scripts from python using the `subprocess` library.

**Comment by pavanramkumar** [comment]
@themantalope I'm going to start work on this now. 

Looking to build a `benchmarks/benchmarks.py` which accepts a list of `environments`, `distrs`, and other `params` and returns a benchmark data frame.

Typical use case would be:

```python
import pyglmnet.benchmarks as benchmarks
environments = ['sklearn', 'pyglmnet', 'R', 'statsmodels']
distrs = ['gaussian', 'binomial', 'poisson']
bench = benchmarks.BenchmarkGLM(environments=environments, distrs=distr, ...)
df = bench.get_benchmarks(Xtrain, ytrain, Xtest, ytest)
```
